src/main/java/games/strategy/triplea/delegate/UnitComparator.java

```diff
@@ -15,148 +14,51 @@
     // this makes it more efficient
     final IntegerMap<Unit> capacityMap = new IntegerMap<>(transports.size() + 1, 1);
     for (final Unit transport : transports) {
       final Collection<Unit> transporting = TripleAUnit.get(transport).getTransporting();
       capacityMap.add(transport, TransportUtils.getTransportCost(transporting));
     }
-    return (t1, t2) -> {
-      final int cost1 = capacityMap.getInt(t1);
-      final int cost2 = capacityMap.getInt(t2);
-      return increasing ? (cost1 - cost2) : (cost2 - cost1);
-    };
+    return Comparator.comparingInt(capacityMap::getInt);
+  }
+
+  private static Comparator<Unit> getDecreasingCapacityComparator(final List<Unit> transports) {
+    return getIncreasingCapacityComparator(transports).reversed();
   }
 
   /**
    * Return a Comparator that will order the specified transports in preferred load order.
    */
   public static Comparator<Unit> getLoadableTransportsComparator(final List<Unit> transports, final Route route,
       final PlayerID player) {
-    final Comparator<Unit> decreasingCapacityComparator = getDecreasingCapacityComparator(transports);
-    final Predicate<Unit> incapableTransportMatch = Matches.transportCannotUnload(route.getEnd());
-    return (u1, u2) -> {
-      final TripleAUnit t1 = TripleAUnit.get(u1);
-      final TripleAUnit t2 = TripleAUnit.get(u2);
-
-      // Check if transport is incapable due to game state
-      final boolean isIncapable1 = incapableTransportMatch.test(t1);
-      final boolean isIncapable2 = incapableTransportMatch.test(t2);
-      if (!isIncapable1 && isIncapable2) {
-        return -1;
-      }
-      if (isIncapable1 && !isIncapable2) {
-        return 1;
-      }
-
-      // Use allied transports as a last resort
-      final boolean isAlliedTrn1 = !t1.getOwner().equals(player);
-      final boolean isAlliedTrn2 = !t2.getOwner().equals(player);
-      if (!isAlliedTrn1 && isAlliedTrn2) {
-        return -1;
-      }
-      if (isAlliedTrn1 && !isAlliedTrn2) {
-        return 1;
-      }
-
-      // Sort by decreasing transport capacity
-      final int compareCapacity = decreasingCapacityComparator.compare(t1, t2);
-      if (compareCapacity != 0) {
-        return compareCapacity;
-      }
-
-      // Sort by decreasing movement
-      final int left1 = t1.getMovementLeft();
-      final int left2 = t1.getMovementLeft();
-      if (left1 != left2) {
-        return left2 - left1;
-      }
-
-      return Integer.compare(t1.hashCode(), t2.hashCode());
-    };
+    return Comparator.comparing(TripleAUnit::get,
+        Comparator.<TripleAUnit, Boolean>comparing(Matches.transportCannotUnload(route.getEnd())::test)
+            .thenComparing(Unit::getOwner, Comparator.comparing(player::equals).reversed())
+            .thenComparing(getDecreasingCapacityComparator(transports))
+            .thenComparing(Comparator.comparingInt(TripleAUnit::getMovementLeft).reversed())
+            .thenComparingInt(Object::hashCode));
```

Comment:
Just noting that I like the way you built the compound comparator in `getUnloadableTransportsComparator()` below rather than the above form.  This seems just a bit cleaner:

```diff
diff --git a/src/main/java/games/strategy/triplea/delegate/UnitComparator.java b/src/main/java/games/strategy/triplea/delegate/UnitComparator.java
index 5d83b42fc..2f5f2ef5a 100644
--- a/src/main/java/games/strategy/triplea/delegate/UnitComparator.java
+++ b/src/main/java/games/strategy/triplea/delegate/UnitComparator.java
@@ -41,12 +41,11 @@ public class UnitComparator {
    */
   public static Comparator<Unit> getLoadableTransportsComparator(final List<Unit> transports, final Route route,
       final PlayerID player) {
-    return Comparator.comparing(TripleAUnit::get,
-        Comparator.<TripleAUnit, Boolean>comparing(Matches.transportCannotUnload(route.getEnd())::test)
-            .thenComparing(Unit::getOwner, Comparator.comparing(player::equals).reversed())
-            .thenComparing(getDecreasingCapacityComparator(transports))
-            .thenComparing(Comparator.comparingInt(TripleAUnit::getMovementLeft).reversed())
-            .thenComparingInt(Object::hashCode));
+    return Comparator.comparing(Matches.transportCannotUnload(route.getEnd())::test)
+        .thenComparing(Unit::getOwner, Comparator.comparing(player::equals).reversed())
+        .thenComparing(getDecreasingCapacityComparator(transports))
+        .thenComparing(TripleAUnit::get, Comparator.comparingInt(TripleAUnit::getMovementLeft).reversed())
+        .thenComparingInt(Object::hashCode);
   }
 
   /**
```